### PR TITLE
0.7.0 rc

### PR DESCRIPTION
**Pull request by davidLeonardi**
Creating new PR for release candidate

### CODE REVIEW

**Comment by johannhof** [comment]
Rebased to remove merge commits :innocent:

**Comment by johannhof** [comment]
@sethx So what exactly does aspect-js solve for us here? Is it worth introducing this library as the only dependency? It doesn't have any tests, we might end up maintaining two libraries...

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "author": "Tristan Edwards <tristan.edwards@me.com> (http://tristanedwards.me)",
   "contributors": [
     "Mikael Carpenter",
+	"Nicolas Charpentier",
```
Comment (johannhof) [inline package.json:8]: nit: wrong indentation

**Comment by davidLeonardi** [comment]
Well, the thing is that Navigator does not emit any events when popping/pushing/and so on.
So the only way to actually get a trigger for those actions is to directy bind to the methods, and that can be done by means of Aspect Oriented Programming. I chose this library because the implementation is a bit less quirky, but it DOES have some bugs.
If somebody knows of a better aspect.before/aspect.after implementation in commonJs please speak up!

**Comment by davidLeonardi** [comment]
https://github.com/hufyhang/aspect-js/issues/1

Let's see how quickly the guy responds.

**Comment by davidLeonardi** [comment]
IF we decide it's not worth the effort, then i'll kill off support for additional events, and that'll be it.

**Comment by charpeni** [comment]
Related to issue #30.

I see this as a breaking change and I think we can't publish that.

My first route is a Splash screen, now with navigationContext listeners the nav bar appears while the following props are true : `hideNavigationBar`, `trans`.

I think i'm not alone with this structure and we should find a solution.

**Comment by davidLeonardi** [comment]
+1 on that. Do you have a solution in mind?

**Comment by charpeni** [comment]
Not for the moment, I will think about it.

**Comment by davidLeonardi** [comment]
Also, #30 is about the deprecation warnings, is that what you really meant?

**Comment by charpeni** [comment]
Yeah, more or less.

#30 is about deprecation, if that listeners are deprecated, we should replace with new version `navigationContext`. But it doesn't work as the previous one.

**Comment by davidLeonardi** [comment]
That's a Navigator deprecation. The events are still emitted in the same
way.. ?

**Comment by davidLeonardi** [comment]
@charpeni i may be a bit thick. WHAT is not working as the previous one? Sorry i just don't get it.
It used to event.emit() , now it event.emit() . what diff?

**Comment by davidLeonardi** [comment]
What's the holdup on this? Not sure what i should do. Please explain.

**Comment by charpeni** [comment]
@sethx 

I am sorry for the delay.

I was saying that since we switch to `navigationContext` the `willfocus` is not emitted on first route. So the route is not changed inside the router component and it's affect my first route, it's a splashscreen page with the followings props : `trans` and `hideNavigationBar`. But it doesn't work since `navigationContext`, we can see the navbar on splashscreen.

To see this issue, just put `trans` and `hideNavigationBar` on your first route. It will not work, but on your second route, it will work.

**Comment by charpeni** [comment]
I just found a solution, I will create a pull request to 0.7.0-rc.

**Comment by ericraio** [comment]
Nice!!

We gotta fix conflicts. :P

**Comment by davidLeonardi** [comment]
Sweet!

**Comment by davidLeonardi** [comment]
So, shall we call 0.7.0 done?

**Comment by charpeni** [comment]
Sure!

**Pull request merged**